lib/ansible/cli/__init__.py

```diff
@@ -575,6 +575,13 @@ def parse(self):
 
         self.options, self.args = self.parser.parse_args(self.args[1:])
 
+        # --limit must receive non-zero length string
+        if hasattr(self.options, 'subset') and  \
+           self.options.subset != C.DEFAULT_SUBSET and \
+           isinstance(self.options.subset, string_types) and \
+           len(self.options.subset) == 0:
+            raise ValueError("ansible-playbook: error: --limit option requires non-zero length string")
```

Comment:
You should raise `AnsibleOptionsError` instead of `ValueError`.  Additionally, drop the `ansible-playbook: error: --` and maybe reword to be "Limit requires a non-zero length string"